### ISSUE TITLE
vg map allows mapping over edges in both directions

### ISSUE THREAD

**Issue by ivargr** [vgteam/vg#1203]
It seems that _vg map_ allows mapping a read over multiple edges that go in different directions. Is this expected behaviour?

Here is a small example to illustrate:
Using this as _graph.json_

    {
    "node": [
        {"id": 1, "sequence": "TTTCCCC"},
        {"id": 2, "sequence": "TTTT"},
        {"id": 3, "sequence": "CCCCTTT"}
    ],
    "edge": [
        {"from": 1, "to": 2},
        {"from": 3, "to": 2, "from_start": true, "to_end": true}
    ]
    }
Then running:

    vg view -Jv graph.json > graph.vg
    vg index -k 11 -g graph.gcsa -x graph.xg graph.vg
    vg map -d graph -s CTTTTC -D -k 1 > aln.gam && vg view -a aln.gam

This gives the following alignment with identity 1.0:
```
{"mapping": [{"position": {"node_id": 1, "offset": 6}, "edit": [{"from_length": 1, "to_length": 1}], "rank": 1}, 
{"position": {"node_id": 2}, "edit": [{"from_length": 4, "to_length": 4}], "rank": 2}, {"position": {"node_id": 3}, 
  "edit": [{"from_length": 1, "to_length": 1}], "rank": 3}]}
```
... which seems to be an alignment starting at node 1, going the _forward_ edge to node 2, then _reversely_ going over the edge from node 3 to 2.

**Comment by ivargr** [comment]
Is anyone able to confirm whether this is a bug or how it is supposed to be?

**Comment by faithokamoto** [comment]
Yeah, that looks valid. The edge 3rev -> 2rev is equivalent to 2 -> 3, so the mapping 1 -> 2 -> 3 is legal.
